### PR TITLE
some message

### PR DESCRIPTION
**Pull request by perronogrande**
- [DEMO LINK](https://perronogrande.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://perronogrande.github.io/layout_product-cards/report/html_report/)

UPDATE 2:
I'm sorry that I took so much of your time to explain me, but now I finally did it, thank you! it even passes the tests now!

### CODE REVIEW

File: src/style.css [hunk 1]
```diff
@@ -0,0 +1,138 @@
+* {
```
Comment (Ihor-Karpyn) [inline src/style.css:1]: імпортуй шрифт https://fonts.google.com/

File: src/index.html [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
-  <title>Product cards</title>
-  <link rel="stylesheet" href="./style.css">
-</head>
-<body>
-  <h1>Product cards</h1>
-</body>
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport"
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
```
Comment (Ihor-Karpyn) [inline src/index.html:6]:
```suggestion
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    >
```

File: src/index.html [hunk 3]
```diff
@@ -1,13 +1,52 @@
...
+      rel="preconnect"
+      href="https://fonts.gstatic.com"
+    >
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+  </head>
+  <body>
+    <div class="main" data-qa="card">
+      <div class="product">
+        <img
+          class="img"
```
Comment (Ihor-Karpyn) [inline src/index.html:20]:
```suggestion
          class="card__img"
```

File: src/index.html [hunk 4]
```diff
@@ -1,13 +1,52 @@
...
+          src="./images/imac.jpeg"
+          alt="imac"
+        >
+      </div>
+      <div class="product-info">
```
Comment (Ihor-Karpyn) [inline src/index.html:25]:
```suggestion
      <div class="card__info">
```

File: src/index.html [hunk 5]
```diff
@@ -1,13 +1,52 @@
...
+        <h2 class="product-name">
```
Comment (Ihor-Karpyn) [inline src/index.html:26]:
```suggestion
        <h2 class="card__name">
```

File: src/index.html [hunk 6]
```diff
@@ -1,13 +1,52 @@
...
+          Моноблок APPLE A1419 iMac<br>27&quot; Retina 5K (MNED2UA/A)
+        </h2>
+        <p class="product-code">
```
Comment (Ihor-Karpyn) [inline src/index.html:29]:
```suggestion
        <p class="card__code">
```

File: src/index.html [hunk 7]
```diff
@@ -1,13 +1,52 @@
...
+          Код товара: 195434
+        </p>
+      </div>
+      <div class="product-rate">
```
Comment (Ihor-Karpyn) [inline src/index.html:33]:
```suggestion
      <div class="card__rate">
```

File: src/index.html [hunk 8]
```diff
@@ -1,13 +1,52 @@
...
+        <div class="stars">
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside-dark"></div>
+        </div>
+        <a class="review-text">
```
Comment (Ihor-Karpyn) [inline src/index.html:41]:
```suggestion
        <a class="card__review">
```

File: src/index.html [hunk 9]
```diff
@@ -1,13 +1,52 @@
...
+          Отзывов:&nbsp;5
+        </a>
+      </div>
+      <div class="price">
+        <h3 class="price-tag">Цена:</h3>
```
Comment (Ihor-Karpyn) [inline src/index.html:46]:
```suggestion
        <h3 class="price__title">Цена:</h3>
```

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,52 @@
...
+        <p class="price-number">69 999 грн</p>
```
Comment (Ihor-Karpyn) [inline src/index.html:47]:
```suggestion
        <p class="price__value">69 999 грн</p>
```

File: src/style.css [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+  margin: 0;
+}
+
+body {
+  font-family: Roboto, sans-serif;
+  width: 200px;
+  height: 404px;
+}
+
+.main {
```
Comment (Ihor-Karpyn) [inline src/style.css:11]:
```suggestion
.card {
```

File: src/index.html [hunk 12]
```diff
@@ -1,13 +1,52 @@
...
+    <link rel="stylesheet" href="./style.css">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link
+      rel="preconnect"
+      href="https://fonts.gstatic.com"
+    >
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+  </head>
+  <body>
+    <div class="main" data-qa="card">
```
Comment (Ihor-Karpyn) [inline src/index.html:17]:
```suggestion
    <div class="card" data-qa="card">
```

File: src/index.html [hunk 13]
```diff
@@ -1,13 +1,52 @@
...
+      <div class="product">
```
Comment (Ihor-Karpyn) [inline src/index.html:18]:
```suggestion
```

File: src/style.css [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+.product {
+  display: flex;
+}
+
+.img {
+  height: 134px;
+  width: 160px;
+  padding-left: 4px;
+  padding-right: 4px;
+  padding-bottom: 40px;
```
Comment (Ihor-Karpyn) [inline src/style.css:27]:
```suggestion
  display: block;
  margin: 0 auto 40px;
```

File: src/style.css [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+.product-name {
+  font-family: Roboto, sans-serif;
+  font-style: normal;
+  font-weight: 500px;
+  font-size: 12px;
+  line-height: 18px;
+  margin: 0;
+  padding-bottom: 2px;
```
Comment (Ihor-Karpyn) [inline src/style.css:44]: use `margin-bottom` instead `padding`

File: src/style.css [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+  font-size: 10px;
+  color: #616070;
+  margin: 0;
+}
+
+.product-rate {
+  display: flex;
+  width: 166px;
+  align-items: center;
+  margin-top: 16px;
```
Comment (Ihor-Karpyn) [inline src/style.css:59]: use `margin-bottom` instead `margin-top`

File: src/style.css [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+  font-weight: 400px;
+}
+
+.price {
+  font-family: Roboto, sans-serif;
+  display: flex;
+  justify-content: space-between;
+  width: 166px;
+  align-items: center;
+  margin-top: 24px;
```
Comment (Ihor-Karpyn) [inline src/style.css:98]: use `margin-bottom` instead `margin-top`

File: src/index.html [hunk 18]
```diff
@@ -1,13 +1,52 @@
...
+        </p>
+      </div>
+      <div class="product-rate">
+        <div class="stars">
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside"></div>
+          <div class="stars-inside-dark"></div>
+        </div>
```
Comment (Ihor-Karpyn) [inline src/index.html:40]:
```suggestion
        <div class="stars stars--4">
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
        </div>
```

File: src/style.css [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+  display: flex;
+  width: 166px;
+  align-items: center;
+  margin-top: 16px;
+}
+
+.stars {
+  display: flex;
+  width: 166px;
+}
```
Comment (Ihor-Karpyn) [inline src/style.css:65]:
```suggestion
.stars {
  display: flex;
}
```

File: src/style.css [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+
+.stars-inside {
+  background-image: url("./images/star-active.svg");
+  width: 16px;
+  height: 16px;
+  background-repeat: no-repeat;
+  margin-top: 1px;
+  margin-right: 2px;
+  margin-left: 2px;
+}
```
Comment (Ihor-Karpyn) [inline src/style.css:75]:
```suggestion
.stars__star {
  height: 16px;
  width: 16px;
  background-image: url("images/star.svg");
  background-repeat: no-repeat;
  background-position: center;
  margin-right: 4px;
}
```

File: src/style.css [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+
+.stars-inside-dark {
+  background-image: url("./images/star.svg");
+  width: 16px;
+  height: 16px;
+  background-repeat: no-repeat;
+  margin-top: 1px;
+  margin-left: 2px;
+}
```
Comment (Ihor-Karpyn) [inline src/style.css:84]:
```suggestion
.stars--1 > .stars__star:nth-child(-n+1),
.stars--2 > .stars__star:nth-child(-n+2),
.stars--3 > .stars__star:nth-child(-n+3),
.stars--4 > .stars__star:nth-child(-n+4),
.stars--5 > .stars__star:nth-child(-n+5) {
  background-image: url("images/star-active.svg");
}
```

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]
https://www.loom.com/share/2a24edb1a0ad45d28b44ede22932850f

File: src/index.html [hunk 22]
```diff
@@ -1,13 +1,50 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+        <a class="card__text">
+          Отзывов:&nbsp;5
+        </a>
+      </div>
+      <div class="price">
```
Comment (aliceblunt8) [inline src/index.html:43]: make price to be an element of card block

File: src/style.css [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+
+body {
+  font-family: Roboto, sans-serif;
+  width: 200px;
+  height: 404px;
+}
+
+.card {
+  padding: 0 16px;
```
Comment (aliceblunt8) [inline src/style.css:14]: add vertical paddings too

File: src/style.css [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+
+.card__info {
+  display: flex;
+  flex-direction: column;
+  margin: 0;
+  margin-bottom: 17px;
+}
+
+.card__name {
+  font-family: Roboto, sans-serif;
```
Comment (aliceblunt8) [inline src/style.css:34]: add to html or body

File: src/style.css [hunk 25]
```diff
@@ -0,0 +1,133 @@
...
+  font-style: normal;
+  font-weight: 500;
+  color: #060b35;
+  font-size: 12px;
+  line-height: 18px;
+  margin: 0;
+  margin-bottom: 5px;
```
Comment (aliceblunt8) [inline src/style.css:41]: combine these 2 rules

File: src/style.css [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+  line-height: 16px;
+  font-weight: 400;
+  color: #060b35;
+}
+
+.price {
+  font-family: Roboto, sans-serif;
+  display: flex;
+  justify-content: space-between;
+  width: 166px;
```
Comment (aliceblunt8) [inline src/style.css:90]: remove width from here

File: src/style.css [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+  font-family: Roboto, sans-serif;
+  font-style: normal;
+  font-weight: 700;
+  font-size: 16px;
+  line-height: 18px;
+  color: #060b35;
+}
+
+.button {
+  font-family: Roboto, sans-serif;
```
Comment (aliceblunt8) [inline src/style.css:114]: add font-family to html

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**